tools/build_script_generator/scons/resources/SConscript.in

```diff
@@ -16,8 +16,8 @@ profile = ARGUMENTS.get("profile", "release")
 env["BUILDPATH"] = join(env["CONFIG_BUILD_BASE"], profile)
 env["BASEPATH"] = abspath(".")
 %% if family == "darwin"
-# Using homebrew gcc-8 on macOS
-env["COMPILERSUFFIX"] = "-8"
+# Using homebrew gcc-9 on macOS
+env["COMPILERSUFFIX"] = "-9"
```

Comment:
Yeah, maybe we can query this and use the latest installed version? I tried compiling with Apple clang, but it's a stdlib disaster for C++17.

Reply:
```suggestion
env["COMPILERSUFFIX"] = env.Detect(["gcc-7", "gcc-8", "gcc-9"])[3:]
```

Reply:
This also needs to be fixed in the CMake file here: https://github.com/modm-io/modm/blob/develop/tools/build_script_generator/cmake/cmake_scripts/configure-gcc.cmake.in#L33